keps/sig-storage/3751-volume-provisioned-io/README.md

```diff
@@ -0,0 +1,1338 @@
+  resources:
+    requests:
+      storage: 64Gi
+```
+
+
+Under the ModifyVolume call, it will pass in the mutable parameters and do the update operation based on the `VolumeAttributesClass` parameters.
+
+
+### 5. Add new operation metrics for ModifyVolume operations
+
+
+```
+- controller modify volume operation duration:
+    - Metric name: storage_operation_duration_seconds{operation_name=modify_volume}
+    - [Optional] Aggregation method: percentile
+    - Components exposing the metric: kube-controller-manager
+- controller modify volume operation errors:
+    - Metric name: storage_operation_errors_total{operation_name=modify_volume}
+    - [Optional] Aggregation method: cumulative counter
+    - Components exposing the metric: kube-controller-manager
+```
+
+### Implementation & Handling Failure
+
+VolumeAttributesClass parameters can be considered as best-effort parameters, the CSI driver should report the status of bad parameters and the volume would fall back to a workable default configuration.
+
+
+### Test Plan
+
+* Basic unit tests for performance and quota system.
+* API conformance tests
+* E2E tests using mock driver to cause failure on create, update and recovering cases
+  * [K8s storage framework](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/storage/testsuites)
+  * [csi-tes](https://github.com/kubernetes-csi/csi-test)
+* Test coverage of quota usage with ResourceQuota and LimitRange
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature in 1.28 behind the VolumeAttributesClass feature gate which is set to a default of false
+- Add basic unit tests
+
+#### Alpha -> Beta
+- Unit tests and e2e tests outlined in design proposal implemented
+
+#### Beta
+
+- Beta in 1.29: Since this feature is an expansion of the existing capacity usage flow, we are going to move this to beta with enhanced e2e and test coverage 
+
+#### GA
+
+- GA in 1.30, an issue board will be created to track issues, and all major issues should be fixed before GA.
+- No users complaining about the new behavior
+
+### Upgrade / Downgrade Strategy
+
+No change in cluster upgrade / downgrade process.
+
+### Version Skew Strategy
```

Comment:
super nit: This sounds more like feature flag combinations than version skew.  It is a very useful table.

Reply:
I copied this from the template, should I rename this session to Feature flag combinations?